backend/box/box.go

```diff
@@ -98,6 +128,93 @@ func init() {
 	})
 }
 
+func getBoxConfig(configFile string) ConfigJSON {
+	jsonData, err := os.Open(configFile)
+	if err != nil {
+		log.Fatalf("Failed to read Box config: %v", err)
+	}
+	byteValue, _ := ioutil.ReadAll(jsonData)
+	var boxConfig ConfigJSON
+	err = json.Unmarshal(byteValue, &boxConfig)
+	if err != nil {
+		log.Fatalf("Failed to parse Box config: %v", err)
+	}
+	return boxConfig
+}
+
+func getClaims(boxConfig ConfigJSON, boxSubType string) *jws.ClaimSet {
+	val, _ := jwtutil.RandomHex(20)
+
+	claims := &jws.ClaimSet{
+		Iss: boxConfig.BoxAppSettings.ClientID,
+		Sub: boxConfig.EnterpriseID,
+		Aud: tokenURL,
+		Iat: time.Now().Unix(),
+		Exp: time.Now().Add(time.Second * 45).Unix(),
+		PrivateClaims: map[string]interface{}{
+			"box_sub_type": boxSubType,
+			"aud":          tokenURL,
+			"jti":          val,
+		},
+	}
+
+	return claims
+}
+
+func getSigningHeaders(boxConfig ConfigJSON) *jws.Header {
+	signingHeaders := &jws.Header{
+		Algorithm: "RS256",
+		Typ:       "JWT",
+		KeyID:     boxConfig.BoxAppSettings.AppAuth.PublicKeyID,
+	}
+
+	return signingHeaders
+}
+
+func getQueryParams(boxConfig ConfigJSON) map[string]string {
+	queryParams := map[string]string{
+		"client_id":     boxConfig.BoxAppSettings.ClientID,
+		"client_secret": boxConfig.BoxAppSettings.ClientSecret,
+	}
+
+	return queryParams
+}
+
+func getDecryptedPrivateKey(boxConfig ConfigJSON) *rsa.PrivateKey {
+
+	block, rest := pem.Decode([]byte(boxConfig.BoxAppSettings.AppAuth.PrivateKey))
+	if len(rest) > 0 {
+		log.Fatalf("Extra data included in key")
+	}
+
+	key, err := pkcs8.ParsePKCS8PrivateKey(block.Bytes, []byte(boxConfig.BoxAppSettings.AppAuth.Passphrase))
+	if err != nil {
+		log.Fatalf("Decrypt failed: %v", err)
+	}
+
+	return key.(*rsa.PrivateKey)
+}
+
+// ConfigJSON defines the shape of a box config.json
```

Comment:
These type definitions maybe should go in `api/types.go`?

Reply:
Yes they definitely should, thanks.